docs/developer/explanation/debian-package-publishing.rst

```diff
@@ -0,0 +1,104 @@
+it performs various checks, including:
+
+- ensuring the signature on the changes file is valid and that the key used
+  is associated with a user who has permission to upload
+  to the target distribution/series/archive
+- ensuring all checksums match
+- confirming that it really is a source upload
+
+..
+    TODO: Create and link to librarian explainer
+
+Once all checks pass, the script uploads the files to the librarian and
+creates a queue entry (``PackageUpload``) in the database.
+If any checks fail the package upload is marked as ``REJECTED``.
+
+If the upload is automatically accepted (e.g. for the devel series),
+the system immediately creates a ``PENDING`` source package publishing history
+entry and creates all necessary binary package build requests.
+Otherwise these are created later when the package is accepted from the queue.
+
+########
+Building
+########
+
+The Launchpad build farm is controlled by the ``buildd-manager`` daemon,
+which manages multiple ``launchpad-buildd`` instances
+for all supported architectures.
+
+For a binary package build, a ``launchpad-buildd`` instance is given a list
+of artifacts to download from the librarian, the source package,
+and the chroot to build it in. The VM then downloads the chroot, unpacks it,
+and uses ``sbuild`` to build the package.
+
+The ``buildd-manager`` queries all ``launchpad-buildd`` instances
+regularly. Once a VM reports that a build is complete, it copies the
+changes file and built artifacts (mostly debs, ddebs, and the buildinfo file)
+off the builder and uploads them as a binary package.
+
+#####################
+Binary package upload
+#####################
+
+The binary packages from the Launchpad builders
+are processed by ``process-upload.py``. This script again performs
+various checks like ensuring that the architecture of the built debs
+matches what was expected and verifying that the checksums match.
+Once all checks pass, the artifacts are copied
+to the librarian and a queue entry is created.
+If any checks fail the package upload is marked as ``REJECTED``.
+
+################
+Queue processing
+################
+
+Binary queue items usually go straight into the ``ACCEPTED`` state, 
+but a build that introduces new binary packages goes into ``NEW``.
+This occurs either because the binary package was newly accepted into
+the archive or due to the package being a kernel package, which are
+always published as new binary packages with the version as a part
+of the binary package name.
+
+When the publisher sees an ``ACCEPTED`` binary queue item,
+it creates new ``PENDING`` binary package publishing history entries.
+
+###########
+Publication
+###########
+
+The archive publisher reads the set of current publication records
+for a component (i.e. main/universe/…) of a suite (which is a combination
+of a series and a pocket like ``noble/proposed`` or ``questing/release``),
+makes sure that these packages are present in the pool and creates
```

Comment:
Can you elaborate what `pool` means here because that is not clear here and the reader might not even know from the lack of special formatting that `pool` is some special thing.